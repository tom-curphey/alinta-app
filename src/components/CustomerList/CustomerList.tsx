import React from 'react';
import { ICustomer } from '../../containers/Customers';

const CustomerList = ({
  customers
}: {
  customers: Array<ICustomer>;
}) => {
  console.log(customers);

  return (
    <section>
      {customers &&
        customers.map((customer, i) => {
          return <li key={i}>{customer.firstName}</li>;
        })}
    </section>
  );
};

export default CustomerList;
