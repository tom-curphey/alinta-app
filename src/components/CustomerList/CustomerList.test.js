import React from 'react';
import { shallow, mount, render } from 'enzyme';
import CustomerList from './CustomerList';

// Shallow rendering only renders one component -> which is one unit
it('Expect to render customer list component', () => {
  const mockCustomers = [
    {
      firstName: 'Tom'
    }
  ];
  expect(
    shallow(<CustomerList customers={mockCustomers} />)
  ).toMatchSnapshot();
});
