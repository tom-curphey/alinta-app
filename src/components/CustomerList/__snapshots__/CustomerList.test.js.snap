// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render customer list component 1`] = `ShallowWrapper {}`;
